src/main.ts

```diff
@@ -632,6 +637,51 @@ ipcMain.handle('data-file:import', async (event) => {
   return path.basename(filePath)
 })
 
+ipcMain.handle(
+  'data-file:load-preview',
+  async (event, fileName: string): Promise<DataFilePreview | null> => {
+    try {
+      const fileType = fileName.split('.').pop()
+      const filePath = path.join(DATA_FILES_PATH, fileName)
+
+      invariant(
+        fileType === 'csv' || fileType === 'json',
+        'Unsupported file type'
+      )
+
+      const { size } = await stat(filePath)
+
+      if (size > FILE_SIZE_PREVIEW_THRESHOLD) {
+        return null
+      }
```

Comment:
Show an error in the UI indicating file is too large to preview, instead of just "no preview available"?

Reply:
or warning 🤔

Reply:
Added a warning/info message. Once the layout update PR is merged, I'll probably also add a grot illustration :D